src/permissions/Caveat.ts

```diff
@@ -6,10 +6,9 @@ import {
   PermissionConstraint,
   RestrictedMethodParameters,
 } from './Permission';
-// This is used in a docstring, but ESLint doesn't notice it.
-/* eslint-disable @typescript-eslint/no-unused-vars */
+// @ts-expect-error This is used in a docstring
```

Comment:
This error specifically is because we reference the type in a JSDoc, we had lines to ignore the warning previously as well, I just changed it to fit with the rules of this repo (required a `ts-expect-error` instead).

The messenging system errors are a bit weird, but it seems to me that they are occurring because we are adding extra allowed actions to the messenger. I.e. access to the controller's own actions etc. So either we have a type error when passing it to the controller or when trying to use it for tests. I can dig up and post the errors in a bit.

Reply:
The messenging system type issues have been fixed in the latest commit

Reply:
Maybe worth disabling `noUnusedLocals` and `noUnusedParameters` in `tsconfig.json` in a separate PR. Those options predate us using ESLint in this repo. They are redundant alongside our ESLint rules.

Reply:
Yeah, I agree. But out of scope

Reply:
#694